### PR TITLE
Change source of Google Workspace `Alternate email` field in checkout

### PR DESCRIPTION
**Pull request by fredrikekelund**
#### Proposed Changes

When buying Google Workspace for an existing domain, we ask users to provide an `Alternate email address` on the checkout page. This detail is needed to set up a Google Workspace customer. The problem is that this field is never pre-populated and users always have to manually enter their details there.

With this PR, we stop relying on the `alternate-email` field that was previously added specifically when users bought Google Workspace for an existing domain. Instead, we use the `email` field, which is what we already display for users who are buying a new domain+Google Workspace. This ensures that the field is correctly pre-populated and that users aren't confronted with an error message as the first thing they see on the checkout page.

This PR is complemented by a back-end patch (D88790-code), **but it does not depend on that change**.

For some additional background, see pbLl1t-1c9-p2. This is also tangentially related to #67420.

| Before | After |
| - | - |
| gws-alternate-email-before | gws-alternate-email-after |

#### Testing Instructions

1. Have an existing domain without an email solution.
2. Proceed to purchase a Google Workspace subscription for that domain.
3. On the checkout page, ensure that you are not immediately confronted with an error message about validation.
4. Also ensure that you see an email address under the `Google Workspace account information` heading. Remember that email address.
5. Purchase the subscription and wait for account setup to complete. 
6. Log in to Google and verify that the alternate email address matches what you saw during checkout. You can do this by looking at `Secondary email` on https://admin.google.com/u/0/ac/accountsettings/profile

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to #

### CODE REVIEW

File: client/components/domains/contact-details-form-fields/index.jsx [hunk 1]
```diff
@@ -513,9 +513,9 @@ export class ContactDetailsFormFields extends Component {
 		return (
 			<div className="contact-details-form-fields__row">
 				<Input
-					label={ this.props.translate( 'Alternate Email Address' ) }
-					{ ...this.getFieldProps( 'alternate-email', {
-						customErrorMessage: this.props.contactDetailsErrors?.alternateEmail,
+					label={ this.props.translate( 'Alternate email address' ) }
+					{ ...this.getFieldProps( 'email', {
+						customErrorMessage: this.props.contactDetailsErrors?.email,
 					} ) }
```
Comment (fredrikekelund) [inline client/components/domains/contact-details-form-fields/index.jsx:519]: We could probably remove all G Suite/Google Workspace related form fields from this file altogether, since it only seems to be used on `/domains/manage/:domain/edit-contact-info/:site_slug` (and not on the checkout page).

However, in the interest of being efficient with our time, I think it might be better to simply leave this as is for now (even no instance of this component seems to use the Google-related form fields).

File: client/my-sites/checkout/composite-checkout/types/wpcom-store-state.ts [hunk 2]
```diff
@@ -531,7 +531,6 @@ export function prepareDomainContactValidationRequest(
 			last_name: details.lastName?.value,
 			organization: details.organization?.value,
 			email: details.email?.value,
-			alternate_email: details.alternateEmail?.value,
```
Comment (sirbrillig) [inline client/my-sites/checkout/composite-checkout/types/wpcom-store-state.ts:534]: If we're removing `alternate_email` from this object and we expect it to never be set, we might want to remove it from the `DomainContactValidationRequest` type (and maybe the `ContactValidationRequestContactInformation`/`ContactValidationResponseMessages` types too?).

Should we also remove it from `formatDomainContactValidationResponse()`?

**Review by sirbrillig — COMMENTED** [review]
This seems like a great idea! The alternate email field was always a bit awkward to handle.

One thing I'm not clear about, though: is the intent to remove the alternate email field from just the Google Workspace contact form, or remove it from everywhere? Because if we want to remove it from everywhere, there's still a lot of places that would need to be modified (grep for `alternateEmail` and `alternate_email` in calypso).

The PR only mentions the Google Workspace form, but I see that it also removes it from `prepareDomainContactValidationRequest()` which is what made me wonder if this was intended to be more far-reaching.

**Comment by delputnam** [comment]
From a domains-related perspective, we don't use the alternate-email field at all, so I think it's safe to remove it everywhere.

**Comment by JessBoctor** [comment]
I did a series of functional tests for this review.

**tl;dr** I found unexpected behavior that if I added contact information in the Google Workspace form on `/email/{customdomain}/purchase/{siteaddress}` and it did not match the contact information from the domain, this contact info appeared to be overwritten in the checkout page by the domain contact information.

From the PR description, I am not sure if this is intentional or not (or within the scope of this test). As a user, if I enter contact information on the sign-up page under the heading `Google Workspace`, I would expect that this same contact information would persist through to `Google Workspace account information` section of the checkout page.

If this is not how this information is intended to work, I think that some clarification is needed in one or both forms.

I got hung up on the contact information not matching between the signup form and checkout. I never made it to completing the Google Workspace account purchase and checking that the reset email matched what I submitted.

I apologize if my test descriptions are repetitive, I just wanted to be as explicit as possible 😀 

Other Notes:
- After doing some digging into where the account information was coming from, it looks like it may be defaulting to the domain contact information somewhere.
- My test account doesn’t have a first name or last name set in the profile.
- The email address in test 3 matched the email address in the domain contact
- My WPcom account email address is an alias address formatted as `myemail+wpcomusername@mydomain.com`
- In tests 1, 2, & 5 the custom domain was set as the primary for the site. In test 3 & 4, the domain was not the primary
- After completing test 4, I could no longer add Google Workspace as an email option to the new domain I had registered. I am not sure if this is because I couldn't verify the email address in the domain contact info (it doesn't exist) or because I had added the product to the cart too many times 😂 
- OS: MacOS 12.5
- Browser: Chrome, version 105.0.5195.125, Non-A8c Chrome profile

<details>
<summary>Test 1</summary>

- Checked out `fix/checkout-autofill-gsuite-alternate-email`

- Accessed via `calypso.localhost:3000`

- Went to `My site > Upgrades > Email` and clicked on `Select` next to the Google Workspace option

- Filled in form for Google Workspace and clicked on `Purchase`

- Worked as expected for the form field; there was no alternative email form. It listed my account email with an option to change it.

- **Unexpected**: The contact information I put into the signup form didn’t carry over to check out. It was overwritten by other information

T1-SignUp

T1-Checkout

</details>
<details>
 <summary>Test 2</summary>

- Checked out `fix/checkout-autofill-gsuite-alternate-email`

- Pulled trunk into my sandbox

- Applied patch `D88790-code`

- Rebased the patch with trunk

- Sandboxed `public-api.wordpress.com`

- Accessed via `calypso.localhost:3000`

- Went to `My site > Upgrades > Email` and clicked on `Select` next to the Google Workspace option

- Filled in form for Google Workspace and clicked on `Purchase`

- Worked as expected for the form field; there was no alternative email form. It listed my account email with an option to change it.

- Got the same response in Checkout, the contact information (first name, last name) I provided didn’t match in Google Workspace account information when I got to the checkout form

T2-SignUp

T2-Checkout

</details>
<details>
 <summary>Test 3</summary>

- Starting from `My Site > Upgrades > Emails` with the same set-up as Test 2

- Clicked on `Select` next to Google Workspace

- Added a custom first name and last name

- Clicked on `Change it `next to my password reset email address and added a different email address

- Clicked on `Purchase`

- **Unexpected**: The Google Workspace account information did not match the information I added in the form **_OR_** my WPcom account information. When I changed the reset email, I used an alias email address of `myemail+testthree@mydomain.com.` It may just be a filtering issue. The email address it had listed was `myemail@mydomain.com`

T3-SignUp

T3-Checkout

</details>
<details>
 <summary>Test 4</summary>

- Using the same setup from Test 2, I purchased a new test domain and Personal plan for a different test site

- I used a different contact name and email address to register the domain, it is unique from previous purchases and my WPcom account

- Went to `My Site > Upgrades > Emails`

- Clicked on `Select` next to Google Workspace

- Added first name and last name, it was unique from the domain contact information and my WPcom account

- The password reset email address was set to my WPcom account email address

- Clicked on `Change it` next to my password reset email address and added a new email address. This one was not an alias

- Clicked on `Purchase`

- **Unexpected**: The Google Workspace account information was written over by the domain contact information
- When I click on `Edit` for the contact information, the domain contact information persists, so I would have to enter the email address again

- If I exit the cart and leave the Google Workspace product in it, then return to the cart, the domain contact information persists

- If I edit the Google Workspace account information in checkout, leave the cart with the product in it and return, the updated contact information persists.

T4-NewPlanDomainPurchase

T4-Checkout-ContactEdit

</details>
<details>
 <summary>Test 5</summary>

- I used the Calypso[ live link](https://calypso.live/?image=registry.a8c.com/calypso/app:build-44680) from `D88790-code`
- Went to `My Site > Upgrades > Emails`

- Clicked on `Select` next to Google Workspace

- Added first name and last name, it was unique from the domain contact information and my WPcom account

- The password reset email address was set to my WPcom account email address

- Clicked on `Change` it next to my password reset email address and added a new email address. This one was not an alias

- Clicked on `Purchase`

- **Unexpected result**: The Google Workspace account information was set to the contact information I added in the Checkout page in the test 4. Not the most recent contact information provided or the domain contact information.

T5-SignUp

T5-Checkout

</details>

File: client/my-sites/checkout/composite-checkout/types/wpcom-store-state.ts (hunk 2 above)
Comment (fredrikekelund) [reply client/my-sites/checkout/composite-checkout/types/wpcom-store-state.ts:534]: Thanks for the review 🙏

You're right, the intent was to remove this field altogether, since it doesn't serve any other purpose than for the Google Workspace checkout form. I've gone ahead and removed the remaining references!

**Comment by fredrikekelund** [comment]
@delputnam perfect, thanks!

@JessBoctor thanks for the detailed tests! I understand why the difference between the first form (email page) and the second form (checkout page) would be confusing when one digs into it. The former is meant to gather details for each individual mailbox (users can buy more than one mailbox in a single go) and the latter is for the overarching Google Workspace "customer". 

One of the benefits of this PR is that users will likely not have to touch the form on the checkout page at all, because if they have previously bought a domain, we should have full contact details on file. So if any users are confused in a similar manner, this should hopefully help alleviate that.

**Review by nightnei — APPROVED** [review]
Everything works as expected and code now looks much clear 👍 LGTM

**Review by nightnei — APPROVED** [review]
👍
